nav2_mppi_controller/include/nav2_mppi_controller/motion_models.hpp

```diff
@@ -52,6 +53,21 @@ class MotionModel
     */
   virtual ~MotionModel() = default;
 
+  /**
+    * @brief Configure motion model on bringup and load required parameters
+    * @param name Name of plugin
+    * @param param_handler Parameter handler object
+    */
+  void on_configure(
```

Comment:
As you mentioned in your comment, the constraint can be violated in the softmax. So, the acceleration constraints should be `get-ed` in the same place and stored alongside the other kino-dynamic constraints -- then passed into the `predict` function to use. That way we keep this all consistent and we have access to the acceleration constraints in the optimizer as well to clip in `applyControlSequenceConstraints` - which as you point out seems necessary

Reply:
I agree with the consistency part that we get all the kino-dynamic constraints stored in the same place. But Optimizer has no use of acceleration constraints, since both `predict` and `applyControlSequenceConstraints` (which 
 calls `motion_model_->applyConstraints`) functions are part of MotionModel class. So, do we need to store it in Optimizer class?

Reply:
applyControlSequenceConstraints also clips the velocities (https://github.com/ros-navigation/navigation2/blob/bf291d704dbb1c2572575dd0620b12f816407233/nav2_mppi_controller/src/optimizer.cpp#L243-L244) shouldn't we also be clipping the accelerations?

Reply:
I think either would be sensible - but I prefer the way I suggested since that's in line with how other things are done for consistency. I think of the `model`'s apply constraints to be unique things to that motion model, and not things we always want to do

Reply:
makes sense.